gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -87,20 +87,28 @@ public Single<PasswordPolicy> create(ReferenceType referenceType, String referen
         log.debug("Create a new password policy named '{}' for {} {}", policy.getName(), referenceType, referenceId);
 
         final var entity = policy.toPasswordPolicy(referenceType, referenceId);
-        // TODO during AM-2893, check if there is existing policies, if not set this one as default
 
-        return passwordPolicyRepository.create(entity)
-                .flatMap(createdPolicy -> {
-                    Event event = new Event(Type.PASSWORD_POLICY, new Payload(createdPolicy.getId(), referenceType, referenceId, Action.CREATE));
-                    return eventService.create(event).flatMap(__ -> Single.just(createdPolicy));
+        return passwordPolicyRepository.findByDefaultPolicy(referenceType, referenceId)
+                .flatMap(__ -> {
+                    entity.setDefaultPolicy(Boolean.FALSE);
+                    return Maybe.just(entity);
```

Comment:
suggestion: why not initializing the value to false by default and only set it to true if the method return an Maybe.empty ? This will avoid unecessary transformation and simplify the code. wdyt ?